### PR TITLE
[ASCII-1334] Create ExpvarServer component

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?
This PR creates the ExpvarServer component

### Describe how to test/QA your changes
Make sure `http://localhost:5000/debug/vars` still returns the expvar values.

### CODE REVIEW

File: comp/agent/bundle_test.go
Comment (misteriaud) [inline comp/agent/bundle_test.go:1]: **💬 suggestion:** Maybe we should add this file to PR #24061‏

**Review by misteriaud — APPROVED** [review]
I left some questions to make better choices on my following PRs, everything LGTM 👍

File: cmd/agent/subcommands/run/command.go [hunk 1]
```diff
@@ -281,6 +283,7 @@ func run(log log.Component,
 		invChecks,
 		statusComponent,
 		collector,
+		expvarserver,
```
[the remark this answers is not in the document]
Comment (ogaca-dd) [reply cmd/agent/subcommands/run/command.go:286]: This is not technically need but I added `expvarserver` to `startAgent` to make sure I didn't miss any calls. As the CI is green I can remove it.

File: comp/agent/expvarserver/expvarserverimpl/expvarserver.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+					Handler: http.DefaultServeMux,
+				}
+				if err := expvarServer.ListenAndServe(); err != nil && !errors.Is(err, http.ErrServerClosed) {
+					deps.Log.Errorf("Error creating expvar server on %v: %v", expvarServer.Addr, err)
+				}
+			}()
+			return nil
+		},
+		OnStop: func(context.Context) error {
+			if expvarServer == nil {
```
Comment (GustavoCaso) [inline comp/agent/expvarserver/expvarserverimpl/expvarserver.go:55]: Should it be `expvarServer != nil` ?

Comment (ogaca-dd) [reply comp/agent/expvarserver/expvarserverimpl/expvarserver.go:55]: Good catch

**Review by GustavoCaso — APPROVED** [review]
LGTM 🎉

**Pull request merged**